### PR TITLE
Add empty function 3 and its docstring

### PR DESCRIPTION
**Pull request by jraza19**
Linked to issue #5

### CODE REVIEW

File: pygtracker/pygtracker.py [hunk 1]
```diff
@@ -23,6 +24,23 @@ def __init__(self):
     # End: Summarize the grades by courses
 
     # Start: Summarise the grades by students
+    def student_ranker(course_id="all", n=10, descending=True):
+        """[Calculates the average grade for a specified number of students and ranks them in descending order for a specific course or for the entire program completed thus far]
```
Comment (tdkhanhvu) [inline pygtracker/pygtracker.py:28]:
```suggestion
        """Calculates the average grade for a specified number of students and ranks them in ascending / descending order for a specific course or for the entire program completed thus far
```

File: pygtracker/pygtracker.py [hunk 2]
```diff
@@ -23,6 +24,23 @@ def __init__(self):
...
+
+        Args:
```
Comment (tdkhanhvu) [inline pygtracker/pygtracker.py:30]: I think we are following numpy style? You can refer to the docstring example in individual assignment one for the format to follow

File: pygtracker/pygtracker.py [hunk 3]
```diff
@@ -23,6 +24,23 @@ def __init__(self):
...
+            course_id (int): [the course ID for which the ranking is calculated for by student.
+            The default option will calculate the ranking for the entire program]. Defaults to "all".
```
Comment (tdkhanhvu) [inline pygtracker/pygtracker.py:32]: Some ways to declare [optional values](https://numpydoc.readthedocs.io/en/latest/format.html) for parameters:

```python
copy : bool, default True
copy : bool, default=True
copy : bool, default: True
```

File: pygtracker/pygtracker.py [hunk 4]
```diff
@@ -23,6 +24,23 @@ def __init__(self):
...
+
+            n(int): [the number of students to rank]. Defaults to 10.
+
+            ascending(bool): [a boolean value to decide if the rank should be in ascending or descending order.
+            The default option is decending order]. Defaults to True.
+
+        Returns:
+            [pandas.DataFrame]: [a dataframe containing the rank of a specified number of students in descending or ascending order
```
Comment (tdkhanhvu) [inline pygtracker/pygtracker.py:40]: I think we agree that it will not return anything?

**Review by tdkhanhvu — CHANGES_REQUESTED** [review]
Should we follow the numpy style here? As it guarantees the readthedocs page will load.

File: pygtracker/pygtracker.py [hunk 5]
```diff
@@ -23,6 +24,23 @@ def __init__(self):
     # End: Summarize the grades by courses
 
     # Start: Summarise the grades by students
+    def student_ranker(course_id="all", n=10, descending=True):
```
Comment (tdkhanhvu) [inline pygtracker/pygtracker.py:27]:
```suggestion
    def rank_students(course_id="all", n=10, descending=True):
```

normally we name the function as a verb:

**Comment by jraza19** [comment]
@tdkhanhvu - thanks for the review - since there are a lot of comments to fix i will close this PR and send a new one.

**Pull request closed without merging**